modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1357,6 +1358,79 @@ impl LightningClientModule {
         }))
     }
 
+    /// Scan unspent incoming contracts for a payment hash that matches a
+    /// tweaked keys in the `indices` vector
+    pub async fn scan_receive_for_user_tweaked(
+        &self,
+        key_pair: KeyPair,
+        indices: Vec<u64>,
+    ) -> anyhow::Result<Amount> {
+        for i in indices {
+            let key_pair_tweaked = tweak_user_secret_key(&self.secp, key_pair, i);
+            if let Ok(amount) = self.scan_receive_for_user(key_pair_tweaked).await {
+                return Ok(amount);
+            }
+        }
+
+        Err(anyhow::anyhow!("No contract found for tweaked keys"))
+    }
+
+    /// Scan unspent incoming contracts for a payment hash that matches a public
+    /// key and claim the incoming contract
+    pub async fn scan_receive_for_user(&self, key_pair: KeyPair) -> anyhow::Result<Amount> {
+        let preimage_key: [u8; 33] = key_pair.public_key().serialize();
+        let preimage = sha256::Hash::hash(&preimage_key);
+        let contract_id = ContractId::from_hash(sha256::Hash::hash(&preimage));
+        self.claim_funded_incoming_contract(
+            key_pair,
+            contract_id,
+            "claiming unspent funded incoming contract",
```

Comment:
`extra_meta` should be exposed to integrators to add custom metadata, not filled with redundant data.